### ISSUE TITLE
Verify links are malware-free

### ISSUE THREAD

**Issue by collinbarrett** [collinbarrett/FilterLists#21]
Using the [VirusTotal API](https://www.virustotal.com/en/documentation/public-api/) (or similar), regularly crawl all links on the site to verify that they are malware-free. This will protect users and prevent potential SEO hits.

**Comment by collinbarrett** [comment]
Partially being done now by the Agent BOT over in #836

**Comment by collinbarrett** [comment]
blocked by #940

**Comment by collinbarrett** [comment]
Maybe integrate https://github.com/funilrys/PyFunceble . Great tip from @DandelionSprout

**Comment by DandelionSprout** [comment]
PyFunceble is slightly difficult to install and keep updated, but it works very well once it has been set up. A simple command like `PyFunceble -uf [File containing full URLs]` should work pretty well.

**Comment by funilrys** [comment]
Hi, there maintainer of PyFuncebke out here!
Thanks for the shootout @collinbarrett and @DandelionSprout ! 

I was looking for discussions about PyFunceble and found this issue. How can I help reduce the difficulties of using, installing, and keeping PyFunceble up-to-date?

Have you tested the [docker images](https://hub.docker.com/u/pyfunceble)? 

Let me know if I can help reduce complexity around PyFunceble!

Have a nice day/night.
Stay safe and healthy.
Nissar

**Comment by DandelionSprout** [comment]
On my end, I had big update problems this spring wherein I had problems updating existing builds, and to get PowerShell and/or Cygwin to recognise the existence of the `PyFunceble` command. My unqualified guess is that it was the result of my trying multiple methods to install and update PyFunceble on my PC (e.g. PIP *and* pure Python), leading to parallel builds of it in various folders that conflicted with each other.

However, in early July I decided to try to delete every single file and folder on my PC that had the word *PyFunceble* in them, and then it went far better to get it to work after a reinstall.

**Comment by funilrys** [comment]
Just a notice for you guys :-) Would you be interested in PyFunceble as a web app (with API call for example) ? 🤔 

I'm starting to think of what to prioritize next and such an idea is my box of ideas. I just need some input about how many people or infrastructure will be interested in that.

**Comment by collinbarrett** [comment]
@funilrys YES!!!!! That'd be awesome. It'd be incredibly helpful to keep the URLs that are on FilterLists clean and valid. I had thought about trying to get an instance of PyFunceble running on the FilterLists server, but just haven't had the time to learn how it works. If there was a simple Web API we could call, that'd be much easier to integrate.

**Comment by funilrys** [comment]
Hi @collinbarrett @DandelionSprout,

I just released beta at https://github.com/PyFunceble/web-worker. 
It's beta because I want to get as much feedback as possible before releasing it as an official stable version.

Feel free to test, report issues if you find some, or create new discussions if something is not clear or if you have some idea that may be useful to others in the future.

Stay safe and healthy.

**Comment by collinbarrett** [comment]
@funilrys nice! would love to test it out. I've been pretty short on time, lately, though, so I'm not sure I'll get around to it real soon.

**Comment by funilrys** [comment]
@collinbarrett Take the time you need! 
Rome was not built in one day - after all. 😉
